joblib/parallel.py

```diff
@@ -37,6 +39,30 @@
 # Environment variables to protect against bad situations when nesting
 JOBLIB_SPAWNED_PROCESS = "__JOBLIB_SPAWNED_PARALLEL__"
 
+# In seconds, should be big enough to hide multiprocessing dispatching
+# overhead.
+# This settings was found by running benchmarks/bench_auto_batching.py
+# with various parameters on various platforms.
+MIN_IDEAL_BATCH_DURATION = .2
+
+# Should not be too high to avoid stragglers: long jobs running alone
+# on a single worker while other workers have no work to process any more.
+MAX_IDEAL_BATCH_DURATION = 2
```

Comment:
I did not really want to treat them as configuration parameter but rather as "universal magic numbers that always work". Otherwise the API is going to explode. Based on benchmarks on various platforms, those values seems to work really well.